pkg/resourcemanager/controller/secret/add.go

```diff
@@ -1,88 +0,0 @@
-// Copyright 2020 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
-//
-// Licensed under the Apache License, Version 2.0 (the "License");
-// you may not use this file except in compliance with the License.
-// You may obtain a copy of the License at
-//
-//      http://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
-
-package secret
-
-import (
-	"context"
-
-	"github.com/go-logr/logr"
-	corev1 "k8s.io/api/core/v1"
-	"k8s.io/apimachinery/pkg/types"
-	"k8s.io/utils/ptr"
-	"sigs.k8s.io/controller-runtime/pkg/builder"
-	"sigs.k8s.io/controller-runtime/pkg/client"
-	"sigs.k8s.io/controller-runtime/pkg/cluster"
-	"sigs.k8s.io/controller-runtime/pkg/controller"
-	"sigs.k8s.io/controller-runtime/pkg/manager"
-	"sigs.k8s.io/controller-runtime/pkg/predicate"
-	"sigs.k8s.io/controller-runtime/pkg/reconcile"
-
-	resourcesv1alpha1 "github.com/gardener/gardener/pkg/apis/resources/v1alpha1"
-	"github.com/gardener/gardener/pkg/controllerutils/mapper"
-	resourcemanagerpredicate "github.com/gardener/gardener/pkg/resourcemanager/predicate"
-)
-
-// ControllerName is the name of the controller.
-const ControllerName = "secret"
-
-// AddToManager adds Reconciler to the given manager.
-func (r *Reconciler) AddToManager(ctx context.Context, mgr manager.Manager, sourceCluster cluster.Cluster) error {
-	if r.SourceClient == nil {
-		r.SourceClient = sourceCluster.GetClient()
-	}
-
-	return builder.
-		ControllerManagedBy(mgr).
-		Named(ControllerName).
-		For(&corev1.Secret{}, builder.WithPredicates(
-			// Only requeue secrets from create/update events with the controller's finalizer to not flood the controller
-			// with too many unnecessary requests for all secrets in cluster/namespace.
-			resourcemanagerpredicate.HasFinalizer(r.ClassFilter.FinalizerName()),
```

Comment:
The `HasFinalizer` func seems to be no longer used: https://github.com/gardener/gardener/blob/bf7bd5ba0914524ca5a04252b4d4f23cc892047d/pkg/resourcemanager/predicate/has_finalizer.go#L28-L49

You could remove it as well.